### PR TITLE
[MODINVSTOR-1217] - Implement Domain Event Production for Campus Create/Update/Delete

### PR DESCRIPTION
**Pull request by Saba-Zedginidze-EPAM**
## Purpose
https://folio-org.atlassian.net/browse/MODINVSTOR-1217

## Approach
Implemented Domain Event Production for Campus Create/Update/Delete

## Screenshots
image
image
1-4300-abbb-042dd57d91da)
image
image
image
image
image

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 ### Features
 * Implement domain event production for location create/update/delete ([MODINVSTOR-1181](https://issues.folio.org/browse/MODINVSTOR-1181))
+* Implement domain event production for institution create/update/delete ([MODINVSTOR-1218](https://issues.folio.org/browse/MODINVSTOR-1218))
```
Comment (BKadirkhodjaev) [inline NEWS.md:11]: This should probably be for Campus

File: src/main/java/org/folio/services/locationunit/CampusService.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  }
+
+  public Future<Response> create(Loccamp loccamp) {
+    return PgUtil.post(CAMPUS_TABLE, loccamp, okapiHeaders, context,
+        PostLocationUnitsCampusesResponse.class)
+      .onSuccess(response ->
+        PostLocationUnitsCampusesResponse
+          .respond201WithApplicationJson(response,
+            PostLocationUnitsCampusesResponse.headersFor201()
+              .withLocation(URL_PREFIX + response)))
```
Comment (BKadirkhodjaev) [inline src/main/java/org/folio/services/locationunit/CampusService.java:63]: As suggested by Serhii, we can remove this custom header

File: src/main/java/org/folio/rest/impl/LocationUnitApi.java [hunk 3]
```diff
@@ -42,7 +44,7 @@ public void getLocationUnitsInstitutions(String query, String totalRecords, int
       cql = StorageHelper.getCql(query, limit, offset, INSTITUTION_TABLE);
     } catch (Exception e) {
       String message = StorageHelper.logAndSaveError(e);
-      asyncResultHandler.handle(Future.succeededFuture(
+      asyncResultHandler.handle(succeededFuture(
```
Comment (BKadirkhodjaev) [inline src/main/java/org/folio/rest/impl/LocationUnitApi.java:47]: We should probably revert these (static class calls) changes that are not related to Campus, in my opinion

**Review by BKadirkhodjaev — APPROVED** [review]

File: src/main/java/org/folio/services/locationunit/CampusService.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+      GetLocationUnitsCampusesResponse.class);
+  }
+
+  public Future<Response> getById(String id) {
+    return PgUtil.getById(CAMPUS_TABLE, Loccamp.class, id, okapiHeaders, context,
+      GetLocationUnitsCampusesByIdResponse.class);
+  }
+
+  public Future<Response> create(Loccamp loccamp) {
+    return PgUtil.post(CAMPUS_TABLE, loccamp, okapiHeaders, context,
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/services/locationunit/CampusService.java:56]: there is no publishing event here, please use the same as in LibraryService or InstitutionalService

File: src/main/java/org/folio/services/locationunit/CampusService.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    if (loccamp.getId() == null) {
+      loccamp.setId(id);
+    }
+
+    return repository.getById(id)
+      .compose(oldLoccamp ->
+        PgUtil.put(CAMPUS_TABLE, loccamp, id, okapiHeaders, context,
+            PutLocationUnitsCampusesByIdResponse.class)
+          .onSuccess(domainEventService.publishUpdated(oldLoccamp))
+          .otherwise(throwable ->
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/services/locationunit/CampusService.java:83]: do we need this otherwise?
For example in LibraryService we don't have it: https://github.com/folio-org/mod-inventory-storage/pull/1029/files#diff-a504c82181f298bf1db3c82e991febfb96b01ed3ed5d5295ea46050e1a2b2fcfR72

Comment (Saba-Zedginidze-EPAM) [reply src/main/java/org/folio/services/locationunit/CampusService.java]: Removed it

File: src/main/java/org/folio/services/locationunit/CampusService.java (hunk 4 above)
Comment (Saba-Zedginidze-EPAM) [reply src/main/java/org/folio/services/locationunit/CampusService.java:56]: Fixed

**Review by SerhiiNosko — APPROVED** [review]

**Review by azizbekxm — APPROVED** [review]

**Review by psmagin — APPROVED** [review]
